Octokit/Clients/Enterprise/EnterpriseAdminStatsClient.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading.Tasks;
+
+namespace Octokit
+{
+    /// <summary>
+    /// A client for GitHub's Enterprise Admin Stats API
+    /// </summary>
+    /// <remarks>
+    /// See the <a href="https://developer.github.com/v3/enterprise/admin_stats/">Enterprise Admin Stats API documentation</a> for more information.
+    ///</remarks>
+    public class EnterpriseAdminStatsClient : ApiClient, IEnterpriseAdminStatsClient
+    {
+        public EnterpriseAdminStatsClient(IApiConnection apiConnection)
+            : base(apiConnection)
+        { }
+
+        /// <summary>
+        /// Gets GitHub Enterprise statistics (must be Site Admin user).
+        /// </summary>
+        /// <remarks>
+        ///https://developer.github.com/v3/enterprise/admin_stats/#get-statistics
+        /// </remarks>
+        /// <returns>The <see cref="AdminStats"/> collection for the requested <see cref="AdminStatsType"/> type.</returns>
+        [SuppressMessage("Microsoft.Globalization", "CA1308:NormalizeStringsToUppercase")]
+        public async Task<AdminStats> GetStatistics(AdminStatsType type)
```

Comment:
@ryangribble thanks for the extra context. I'm not feeling _strongly_ either way at this stage, I was just reflecting on how we don't really have a good example of another API that passes in an enum like this.

For most of the existing API endpoints the parameters are bundled up into a single class - the `XYZRequest` overloads - but that's not really useful here. Whether you go with one method taking an enum of N values, or N methods each representing one endpoint, the only differences beyond the semantics are:

 - does one or the other make more sense to the consumer?
 - does one or the other make the implementation code easier or harder?

For me, I think it's a little weird to always get an `AdminStats` response model back, even if I only asked for `users`. After the first time it failed, and I'd go "ah, I want the `Users` property" and update the code, but I'd love to :fire: the friction off completely.

Do you think we'd get reuse from the existing response models you've created if we went down the "one method per endpoint" scenario? It's going to introduce more methods, but I think they'll be a lot more succinct (and simplify the tests a bit) if we can leverage the serialization cheats we've got to just get back the data we need.